packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/addDevice/add-device.controller.js

```diff
@@ -11,12 +12,21 @@ export default class SoftphoneAddDeviceController {
   }
 
   generateLink() {
-    this.softphoneService
-      .generateLink(
+    return this.softphoneService
+      .setDeviceName(
         this.$stateParams.billingAccount,
         this.$stateParams.serviceName,
         this.model.name,
+        this.deviceId,
       )
+      .then(({ deviceId }) => {
```

Comment:
If your input name doesn't change, in this case you don't request "put deviceId" when you ask new enroll. You can use a condition with form.$pristine or form.$dirty.

Reply:
I will close this one in favor of this one: https://github.com/ovh/manager/pull/11569

(separated logic + optimize one put call)